### PR TITLE
fix --host-config-dir, don't download host path from docker container

### LINKED ISSUE #11606: Failure to start a oc cluster when specifying --host-config-dir
Failure to start a oc cluster when specifying --host-config-dir
##### Steps To Reproduce

```
[INFO] Running a new cluster
oc cluster up --public-hostname '127.0.0.1'                 --host-data-dir '/Users/jmorales/.oc/profiles/origin-14/data'                 --host-config-dir '/Users/jmorales/.oc/profiles/origin-14/config'                 --routing-suffix 'apps.lcup'                 --use-existing-config
-- Checking OpenShift client ... OK
-- Checking Docker client ... OK
-- Checking Docker version ... OK
-- Checking for existing OpenShift container ... OK
-- Checking for openshift/origin:v1.4.0-alpha.0 image ... OK
-- Checking Docker daemon configuration ... OK
-- Checking for available ports ... OK
-- Checking type of volume mount ...
   Using Docker shared volumes for OpenShift volumes
-- Creating host directories ... OK
-- Finding server IP ...
   Using public hostname IP 127.0.0.1 as the host IP
   Using 127.0.0.1 as the server IP
-- Starting OpenShift container ... ERROR: Error reading next tar header: API error (404): {"message":"lstat /var/lib/docker/aufs/mnt/16fa32634472497de351bcedf07fa8ff7b693802bfb0ceae4816ea43c0fda490/Users/jmorales/.oc/profiles/origin-14/config: no such file or directory"}
ERROR: Error extracting tar stream

   Creating initial OpenShift configuration
ERROR: Error reading next tar header: API error (404): {"message":"lstat /var/lib/docker/aufs/mnt/db6120806819e72018e87a5ea7820ae2c0dccbf42ed92631bec7c330086776e0/Users/jmorales/.oc/profiles/origin-14/config: no such file or directory"}
ERROR: Error extracting tar stream
FAIL
   Error: could not copy OpenShift configuration
   Caused By:
     Error: API error (404): {"message":"lstat
/var/lib/docker/aufs/mnt/db6120806819e72018e87a5ea7820ae2c0dccbf42ed92631bec7c330086776e0/Users/jmorales/.oc/profiles/origin-14/config:
no such file or directory"}
```
##### Version

```
oc v1.4.0-alpha.0+ca101d7-816
kubernetes v1.4.0+776c994
features: Basic-Auth
```

This is the docker for mac version I'm using (1.12.2-rc1-beta27):

```
$docker version
Client:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        Tue Sep 27 23:38:15 2016
 OS/Arch:      darwin/amd64
 Experimental: true

Server:
 Version:      1.12.2-rc1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   45bed2c
 Built:        Tue Sep 27 23:38:15 2016
 OS/Arch:      linux/amd64
 Experimental: true
```

### PR DESCRIPTION
**Pull request by jim-minter**
fixes #11606

### CODE REVIEW

File: pkg/bootstrap/docker/openshift/helper.go [hunk 1]
```diff
@@ -434,13 +434,13 @@ func masterHTTPClient(localConfig string) (*http.Client, error) {
...
-func (h *Helper) copyConfig(hostDir string) (string, error) {
+func (h *Helper) copyConfig() (string, error) {
 	tempDir, err := ioutil.TempDir("", "openshift-config")
 	if err != nil {
 		return "", err
 	}
-	glog.V(1).Infof("Copying from host directory %s to local directory %s", hostDir, tempDir)
-	if err = h.hostHelper.DownloadDirFromContainer(hostDir, tempDir); err != nil {
+	glog.V(1).Infof("Copying host config to local directory %s", tempDir)
+	if err = h.hostHelper.DownloadDirFromContainer("/var/lib/origin/openshift.local.config", tempDir); err != nil {
```
Comment (mfojtik) [inline pkg/bootstrap/docker/openshift/helper.go]: make this path const

Comment (jim-minter) [inline pkg/bootstrap/docker/openshift/helper.go]: done

File: pkg/bootstrap/docker/openshift/helper.go [hunk 2]
```diff
@@ -243,7 +244,7 @@ func (h *Helper) Start(opt *StartOptions, out io.Writer) (string, error) {
 	skipCreateConfig := false
 	if opt.UseExistingConfig {
 		var err error
-		configDir, err = h.copyConfig(opt.HostConfigDir)
+		configDir, err = h.copyConfig()
```
Comment (bparees) [inline pkg/bootstrap/docker/openshift/helper.go]: why are we assuming a hardcoded config path instead of using the user provided config path?

**Comment by bparees** [comment]
@jim-minter i need more context around the root cause of the issue and how this fixes it

**Comment by bparees** [comment]
@jim-minter or you can just get @csrwng to sign off on it :)

**Comment by jim-minter** [comment]
@bparees I changed the semantics of the container file download function in a previous commit and missed this fixup.  Previously the function could download arbitrary files from the docker host, therefore needed to be told wherever /var/lib/origin/openshift.local.config was bound to (opt.HostConfigDir).  Now it downloads only from within the context of the container filesystem, and "/var/lib/origin/openshift.local.config" is constant.

**Comment by jim-minter** [comment]
So opt.HostConfigDir is respected in the binds of the container, rather than in the filepath that is requested for download.

**Comment by bparees** [comment]
got it. thanks.
[merge]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/10773
[test]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/11649
[test]

**Comment by csrwng** [comment]
https://github.com/openshift/origin/issues/10988
[test]

**Comment by bparees** [comment]
flake https://github.com/openshift/origin/issues/10773
[merge]

**Comment by jorgemoralespou** [comment]
@jim-minter @csrwng thanks

**Pull request merged**